src/sqlparser/src/parser.rs

```diff
@@ -342,6 +338,10 @@ impl Parser {
 
             expr = self.parse_infix(expr, next_precedence)?;
         }
+        // Cast off nested expression to avoid interface by parentesis.
+        while let Expr::Nested(expr1) = expr {
+            expr = *expr1;
+        }
```

Comment:
Can we safely have all subexprs to be without parenthesis? Is there any special case(s) to consider?

Reply:
I think I shall just keep unnesting local to negative expressions. There seem to be many areas nested expressions are used, and it's not entirely clear which are necessary / redundant.

Reply:
If some of the existing nesting is incorrect etc... we can fix in separate PR.

Reply:
reverted